app/helpers/katello/katello_urls_helper.rb

```diff
@@ -37,11 +37,38 @@ def subscription_manager_configuration_url(host = nil, rpm = true, hostname: nil
       "#{prefix}/pub/#{config}"
     end
 
-    apipie :method, 'Generates an absolute path to the file' do
+    apipie :method, 'Generates an absolute path to the file for liveimg kickstart templates' do
       required :content_path, String, desc: "Relative path to the file or it's name"
       optional :schema, String, desc: 'Optional URL schema for the content source', default: 'http'
       optional :content_type, String, desc: 'Content type', default: 'repos'
       returns String, desc: 'Absolute path to a file'
     end
+    def repository_url(content_path, _content_type = nil, schema = 'http')
+      return content_path if content_path =~ %r|^([\w\-\+]+)://|
+      url = if @host.content_source
+              "#{schema}://#{@host.content_source.hostname}"
+            else
+              foreman_settings_url(schema)
+            end
+      content_path = content_path.sub(%r|^/|, '')
+
+      lifecycle_environment = nil
+      if @host.content_views.present?
+        if @host.content_views.to_a.intersection(::Katello::ContentView.default).present?
```

Comment:
There's a method that can simplify this:

https://github.com/Katello/katello/blob/679f08e6bffad977e0ffc70be4fc78e76582932a/app/models/katello/host/content_facet.rb#L128-L133
```suggestion
      if @host.default_environment?
```

Reply:
Fixed!